src/components/SearchInput/SearchInput.constants.ts

```diff
@@ -1,12 +1,27 @@
+import { IconScale } from '../Icon';
+
 const CLASS_PREFIX = 'md-search-input';
 
-const DEFAULTS = {};
+const HEIGHTS = {
+  28: 28,
+  32: 32,
+};
+
+const SEARCH_ICON_HEIGHT_MAPPING: Record<number, IconScale> = {
+  28: 12,
+  32: 16,
+};
+
+const DEFAULTS = {
+  HEIGHT: HEIGHTS[32],
+};
```

Comment:
Here just use the number as the default height
Also I recommend to add the `as const`, see the reason above

```suggestion
const DEFAULTS = {
  HEIGHT: 32,
} as const;
```